redfish-core/lib/ethernet.hpp

```diff
@@ -1103,20 +1103,92 @@ inline void
+            {
+                for (const auto& interface : objpath.second)
+                {
+                    if (interface.first == "xyz.openbmc_project.Network.IP")
+                    {
+                        auto type = std::find_if(interface.second.begin(),
+                                                 interface.second.end(),
+                                                 [](const auto& property) {
+                            return property.first == "Type";
+                        });
+                        const std::string* typeStr =
+                            std::get_if<std::string>(&type->second);
+
+                        if (typeStr == nullptr ||
+                            (*typeStr !=
+                             "xyz.openbmc_project.Network.IP.Protocol.IPv4"))
+                        {
+                            continue;
+                        }
+                        auto origin = std::find_if(interface.second.begin(),
+                                                   interface.second.end(),
+                                                   [](const auto& property) {
+                            return property.first == "Origin";
+                        });
+                        const std::string* originStr =
+                            std::get_if<std::string>(&origin->second);
+
+                        if (originStr == nullptr ||
+                            (*originStr !=
+                             "xyz.openbmc_project.Network.IP.AddressOrigin.Static"))
+                        {
+                            continue;
+                        }
+
+                        std::string id =
+                            objpath.first.str.substr(ipPathStart.size());
+                        deleteIPv4(ethifaceId, id, asyncResp);
+                    }
+                }
+            }
+        }
+        },
+        "xyz.openbmc_project.Network", "/xyz/openbmc_project/network",
+        "org.freedesktop.DBus.ObjectManager", "GetManagedObjects");
+}
+
 inline void setDHCPEnabled(const std::string& ifaceId,
                            const std::string& propertyName, const bool v4Value,
                            const bool v6Value,
                            const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
     const std::string dhcp = getDhcpEnabledEnumeration(v4Value, v6Value);
     crow::connections::systemBus->async_method_call(
-        [asyncResp](const boost::system::error_code ec) {
+        [asyncResp, ifaceId, v4Value](const boost::system::error_code& ec) {
         if (ec)
         {
             BMCWEB_LOG_ERROR << "D-Bus responses error: " << ec;
             messages::internalError(asyncResp->res);
             return;
         }
+        // TODO this is workaround needed to avoid ipv4 static and DHCP
+        // address co-existence This behaviour should be fixed in
+        // backend networkd D-bus application
```

Comment:
Why can't we fix this in the network D-Bus app?

Reply:
networkd design needs to be changed to fix this, have tried to fix but seeing issues in the networkd
we have started discussions for networkd design changes with upstream maintainers.

Reply:
I would really prefer we fix this in networkd
bmcweb has enough hacks

Reply:
Can we fix in downstream networkd ? Work those upstream?

Reply:
this PR clears at least 5 critical sev1 network defects

Reply:
Upstream wouldn’t allow this in bmcweb, I’m struggling why we should allow this downstream in bmcweb. How much effort is the networkd change? A few days or weeks? Can we wait until we make the changes in networkd?

Reply:
If you discuss Monday at scrum of scrums overflow and decide this is the correct course, I’m good but I really think we should just fix the right way in networkd now.

Reply:
Gunnar 
upstream networkd design completely changed and as part 1050 rebase we pulled master code which allows both ipv4 dhcp and static IP co-existence & networkd broken completely, we need to figure out networkd fix yet..
as I mentioned we need to pull this workaround for sometime to unblock fvt